### DISCUSSION TITLE
Use specific mirror and look for updates using library/{{{packageName}}} for official images not working

### DISCUSSION THREAD

**Discussion by nilson-aguiar** [renovatebot/renovate#22961] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.141.2

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I have a [repo](https://github.com/nilson-aguiar/renovate-mirror-test) containing an `stg` folder containing a `redis.yaml` file.

I specifically want to set "https://mirror.gcr.io" as my default registryURL. For the official images to be evaluated and I added an extra rule to change them to the same mirror, I've used [this info from the docs](https://docs.renovatebot.com/configuration-options/#replacementnametemplate) to create it. You can see it in the `renovate.json` file on the repo above.

From the logs, I could notice that instead of looking for tags and adding the library URL, it's reaching the mirror like this:

```json
{
  "level": 10,
  "repository": "nilson-aguiar/renovate-mirror-test",
  "url": "https://mirror.gcr.io/v2/redis/tags/list?n=10000",
  "msg": "Authorization disabled",
  "v": 0
}
```

I'd expect it to reach the mirror using this URL instead -> https://mirror.gcr.io/v2/library/redis/tags/list?n=10000

Not sure if there is another way to do what I'm looking for, but it was what I could find from the docs.

### Relevant debug logs

<details><summary>Logs</summary>

```
{
  "repository": "nilson-aguiar/renovate-mirror-test",
  "allRequests": [
    "POST https://api.github.com/graphql 200 249 0",
    "POST https://api.github.com/graphql 200 772 0",
    "GET https://api.github.com/repos/nilson-aguiar/renovate-mirror-test/issues/11 200 229 0",
    "GET https://api.github.com/repos/nilson-aguiar/renovate-mirror-test/issues/11 200 221 0",
    "GET https://api.github.com/repos/nilson-aguiar/renovate-mirror-test/pulls?per_page=100&state=all&sort=updated&direction=desc&page=1 200 461 0",
    "GET https://mirror.gcr.io/v2/redis/tags/list?n=10000 200 410 0"
  ],
  "requestHosts": {
    "api.github.com": [
      {
        "method": "post",
        "url": "https://api.github.com/graphql",
        "queueDuration": 0,
        "duration": 249,
        "statusCode": 200
      },
      {
        "method": "post",
        "url": "https://api.github.com/graphql",
        "queueDuration": 0,
        "duration": 772,
        "statusCode": 200
      },
      {
        "method": "get",
        "url": "https://api.github.com/repos/nilson-aguiar/renovate-mirror-test/issues/11",
        "queueDuration": 0,
        "duration": 229,
        "statusCode": 200
      },
      {
        "method": "get",
        "url": "https://api.github.com/repos/nilson-aguiar/renovate-mirror-test/issues/11",
        "queueDuration": 0,
        "duration": 221,
        "statusCode": 200
      },
      {
        "method": "get",
        "url": "https://api.github.com/repos/nilson-aguiar/renovate-mirror-test/pulls?per_page=100&state=all&sort=updated&direction=desc&page=1",
        "queueDuration": 0,
        "duration": 461,
        "statusCode": 200
      }
    ],
    "mirror.gcr.io": [
      {
        "method": "get",
        "url": "https://mirror.gcr.io/v2/redis/tags/list?n=10000",
        "queueDuration": 0,
        "duration": 410,
        "statusCode": 200
      }
    ]
  },
  "msg": "full stats",
  "time": "2023-06-24T16:15:41.655Z",
  "v": 0
}
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by viceice** [comment]
use registry aliases instead

**Reply by nilson-aguiar** [reply]
Would you be able to point me to an example? Tried checking the docs and I wasn't able to link `registryAlias` with a `docker datasource` or a `regex manager`

**Reply by viceice** [reply]
for regex manager i would recommend to simply add the required prefix to the package name
